python/ray/train/xgboost/config.py

```diff
@@ -37,28 +41,93 @@ class XGBoostConfig(BackendConfig):
     def train_func_context(self):
         @contextmanager
         def collective_communication_context():
-            with CommunicatorContext():
+            with CommunicatorContext(**_get_xgboost_args()):
                 yield
 
         return collective_communication_context
 
     @property
     def backend_cls(self):
         if self.xgboost_communicator == "rabit":
-            return _XGBoostRabitBackend
+            return (
+                _XGBoostRabitBackend
+                if Version(xgboost.__version__) >= Version("2.1.0")
+                else _XGBoostRabitBackend_pre_xgb210
+            )
 
         raise NotImplementedError(f"Unsupported backend: {self.xgboost_communicator}")
 
 
 class _XGBoostRabitBackend(Backend):
     def __init__(self):
-        self._tracker = None
+        self._tracker: Optional[RabitTracker] = None
+        self._wait_thread: Optional[threading.Thread] = None
+
+    def _setup_xgboost_distributed_backend(self, worker_group: WorkerGroup):
+        # Set up the rabit tracker on the Train driver.
+        num_workers = len(worker_group)
+        rabit_args = {"n_workers": num_workers}
+        train_driver_ip = ray.util.get_node_ip_address()
+
+        # NOTE: sortby="task" is needed to ensure that the xgboost worker ranks
+        # align with Ray Train worker ranks.
+        # The worker ranks will be sorted by `dmlc_task_id`,
+        # which is defined below.
+        self._tracker = RabitTracker(
+            n_workers=num_workers, host_ip=train_driver_ip, sortby="task"
+        )
+        self._tracker.start()
+
+        # The RabitTracker is started in a separate thread, and the
+        # `wait_for` method must be called for `worker_args` to return.
+        self._wait_thread = threading.Thread(target=self._tracker.wait_for, daemon=True)
+        self._wait_thread.start()
```

Comment:
Is this actually a regression of the `RabitTracker` or an improvement that XGBoost want to continue support?

Reply:
I think 2.1+ requires `wait_for` to be called. otherwise the process group init will hang forever.

previously, `tracker.start` was all that's needed. I updated this by following an example in their tests.

Reply:
It's a bit problematic that these developerAPIs are not that stable, which could result in us needing to have multiple versions to support. But, this would be the case with any integration, so it's just something we have to put up with, or complain to xgboost developers about. ideally we can add some ray test to their CI.